### ISSUE TITLE
about training

### ISSUE THREAD

**Issue by xuyuxiu83** [bartn8/vppstereo#3]
Thanks for providing us with the evaluation codes. May I know that if it is feasible to modify a training code according to your evaluation code？

**Comment by xuyuxiu83** [comment]
I am very eager to hear your early reply and it would be greatly appreciated!!!

**Comment by bartn8** [comment]
Hi Xu,

I'm glad that you are interested in our project.

Yes, you can easily create a training procedure from the evaluation code: you can apply the virtual pattern (VPP) using the function vpp inside ```vpp_standalone.py``` before passing the stereo images to the stereo model.
You don't have to worry about differentiability since VPP is applied to the input and does not interfere with the stereo model.
I suggest you also include some type of "VPP augmentations", such as random depth hints density, random patch size, and so on.

Best,
Luca

**Comment by xuyuxiu83** [comment]
> Hi Xu,
> 
> I'm glad that you are interested in our project.
> 
> Yes, you can easily create a training procedure from the evaluation code: you can apply the virtual pattern (VPP) using the function vpp inside `vpp_standalone.py` before passing the stereo images to the stereo model. You don't have to worry about differentiability since VPP is applied to the input and does not interfere with the stereo model. I suggest you also include some type of "VPP augmentations", such as random depth hints density, random patch size, and so on.
> 
> Best, Luca

Thank you for your prompt reply! I still have a few questions that I would greatly appreciate your insights on, if you're able to share:

During fine-tuning, was the loss function based on the original stereo project, or did you use the weighted implementation defined in your own losses.py?

In your previous response, you mentioned random patch selection and random depth hint density—would you be able to provide a rough range or guideline for those?

Are there any other important details I should be aware of beyond the ones mentioned above?

**Comment by bartn8** [comment]
Sorry, I forgot to reply.

For each stereo model we used the original training protocol.
For patch size I suggest to use the set {1,3,5,7}, while for density {0,1,2,3,4,5}% 

I think this is a good starting point.

Best,
Luca
